ResearchKit/Common/ORKFormStepViewController.m

```diff
@@ -811,18 +812,29 @@ - (void)hideSections {
                 [_hiddenFormItems addObject:formItem];
                 if (cellItem) {
                     [newHiddenCellItems addObject:cellItem];
+                    if (![_hiddenCellItems containsObject:cellItem]) {
+                        sectionHasChanges = YES;
+                    }
                 }
             } else {
                 if (cellItem && currentRowIndex == NSNotFound) {
                     [pendingRowInsertions addObject:[NSIndexPath indexPathForRow:newRows.count inSection:newSections.count]];
                 }
                 if (cellItem) {
                     [newRows addObject:cellItem];
+                    if ([_hiddenCellItems containsObject:cellItem]) {
+                        sectionHasChanges = YES;
+                    }
                 }
                 hideSection = NO;
             }
         }
         
+        /*
+         Due to cell re-creation to fix rounded corners with changes, the tableview resigns any first responder so if in the middle of
+         editing a text field, we need to capture and re
```

Comment:
Comment text is truncated

Reply:
Removed - from a previous idea to solve issue